data/src/main/java/org/apache/iceberg/data/parquet/GenericParquetReaders.java

```diff
@@ -366,6 +365,15 @@ public OffsetDateTime read(OffsetDateTime reuse) {
     }
   }
 
+  private static class TimeReader extends PrimitiveReader<LocalTime> {
+    private TimeReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public LocalTime read(LocalTime reuse) { return LocalTime.MIN.plus(column.nextLong(), ChronoUnit.MICROS); }
```

Comment:
In other places, we use `LocalTime.ofNanoOfDay(decoder.readLong() * 1000L)`, so I'm not sure that this implementation is correct.

Could you write a test that validates the conversion from microseconds to a time value? You can convert a time string to a microseconds value using `Literals.of("10:11:12.131415").to(TimeType.get()).value()`.

Reply:
Sure thing - will change this and write that test.

Reply:
@rdblue I did update to the alternative implementation suggested, but I actually think conversion is covered by existing tests, since I added a line to `DataTest` - there was already a case match in `RandomGenericData.generatePrimitive` for the `TIME` type. Existing tests write this to the database as a `Long` and then read the result as `Time`. Do you think that's enough?

Reply:
I think that test will work. Originally, I thought that it wasn't actually testing the conversion, only that writing a time and reading it back produced the same value. But that's okay because we are at least validating that the read path does the opposite of the write path.